### DISCUSSION TITLE
Use 17 Lands data for STX boosters

### DISCUSSION THREAD

**Discussion by wrmacrae** [Senryoku/Draftmancer#453] [General]
As a mostly-better fix for #271 we should just use the data available here https://www.17lands.com/public_datasets to generate STX boosters as long as there are default settings with Ignore Collections set. Going forward we'll want a way to leverage data for future sets and bring in the growing data sets, but in the immediate term, this means collation can be much more realistic without as much set-specific code.

As a side note, I don't think that 17 Lands data can deprecate our collation code, because it's important to support custom rules, old sets, collections, and spoiler-season-drafts (i.e. after the set is spoiled but before Arena release). But I do think it allows one of the most common case experiences to be more authentic to Arena without as much recurring work from us.

**Comment by Senryoku** [comment]
There's already an alternate collation method (called "PaperBoosters") based on https://github.com/taw/magic-sealed-data. The color balancing is still up to us, but we can still iterate on it if necessary, like you proposed in #271.
(Granted, I haven't updated it for STX yet... And I didn't explore their "-arena" versions of sets.)

How would you like to integrate 17lands data exactly? I'm not sure there's a straightforward way to turn these 5GB draft logs into a collation algorithm ^^". It's definitely interesting to test theories about Arena collation though, it would be cool to have an easy way to explore/query these logs.

**Comment by wrmacrae** [comment]
Interesting. It doesn't look like taw's paper one for STX is accurate, unless WotC drastically changed paper collation rules since IKO. But I see those ones for "-arena" which look mostly right. Like you said, that would still leave us to interpret `"balance_colors": true` which feels like the hard part here.

As for 17Lands, I was imagining we can pre-process the logs to get a list of full boosters only, and possibly further compress that down to shorter IDs if necessary. Even though the full uncompressed log is 5GB, I feel like we should be able to get all of the full boosters (looks like ~150k of them from the premier drafts) into tens or hundreds of megabytes of memory, even before going too crazy with optimizations. That's a big enough number that we can pick a random pack from the data instead of collating without noticeable repeats, but a small enough number that the data should be fine server-side.

**Comment by pmbarrett314** [comment]
One insight that has come from the 17lands data is that there is still an uncommon print run. I think behind the scenes they've just made the run much, much longer, but all I know for sure is that there are exactly 1600 possibilities for the 3 uncommons in an Arena Draft Booster of STX. If y'all are interested in those, you can find them [here](https://pastebin.com/V1tguYx1).
